File: OCH_Tao/tessoku-book/B06.py
from itertools import accumulate
N = int(input())
A = ([0]+list(accumulate(map(int, input().split()))))
Q = int(input())
for i in range(Q):
  L, R = map(int, input().split())
  X = A[R]-A[L-1]
  if X > (R-L+1)-X:
    print("win")
  elif X < (R-L+1)-X:
    print("lose")
  else:
    print("draw")

File: OCH_Tao/tessoku-book/B06.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
input.shift();
const A = input.shift().split(" ").map(x => parseInt(x, 10));
let X = [0];
for (const i of A) {
  X.push(X[X.length - 1] + i);
}
input.shift();
for (const i of input) {
  const [L, R] = i.split(" ").map(x => parseInt(x, 10));
  const Y = X[R] - X[L - 1];
  console.log(Y > (R - L + 1) - Y ? "win" : Y < (R - L + 1) - Y ? "lose" : "draw");
}